app/src/main/java/org/mozilla/fenix/settings/address/AddressUtils.kt

```diff
@@ -0,0 +1,158 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.address
+
+import androidx.annotation.StringRes
+import androidx.annotation.VisibleForTesting
+import mozilla.components.concept.storage.Address
+import org.mozilla.fenix.R
+
+internal const val DEFAULT_COUNTRY = "US"
+
+/**
+ * Value type representing properties determined by the country used in an [Address].
+ * This data is meant to mirror the data currently represented on desktop here:
+ * https://searchfox.org/mozilla-central/source/toolkit/components/formautofill/addressmetadata/addressReferences.js
+ *
+ * This can be expanded to included things like a list of applicable states/provinces per country
+ * or the names that should be used for each form field.
+ *
+ * Note: Most properties here need to be kept in sync with the data in the above desktop
+ * address reference file in order to prevent duplications when sync is enabled. There are
+ * ongoing conversations about how best to share that data cross-platform, if at all.
+ * Some more detail: https://bugzilla.mozilla.org/show_bug.cgi?id=1769809
+ *
+ * Exceptions: [displayName] is a local property and stop-gap to a more robust solution.
+ *
+ * @property key The country code used to lookup the address data. Should match desktop entries.
+ * @property displayName The name to display when selected.
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+internal data class Country(
+    val key: String,
```

Comment:
should this just be called "countryCode"?

Reply:
This was intended to mirror the address reference data on desktop, but that does seem more clear for now.